graveler/committed/sstable/cache.go

```diff
@@ -1,14 +1,103 @@
 package sstable
 
 import (
+	"fmt"
+
 	"github.com/cockroachdb/pebble/sstable"
+	lru "github.com/treeverse/lakefs/cache"
 	"github.com/treeverse/lakefs/graveler/committed"
+	"github.com/treeverse/lakefs/pyramid"
 )
 
 //go:generate mockgen -source=cache.go -destination=mock/cache.go -package=mock
 
+type Derefer lru.Derefer
+
+type opener func(namespace string, id committed.ID) (Item, error)
+
 type cache interface {
-	// derefer is the type of a function that returns an object to the cache, possibly causing
-	// its eviction.
-	GetOrOpen(namespace string, id committed.ID) (reader *sstable.Reader, derefer func() error, err error)
+	GetOrOpen(namespace string, id committed.ID) (*sstable.Reader, Derefer, error)
+}
+
+type lruCache struct {
+	c    lru.CacheWithDisposal
+	open opener
+}
+
+// item holds an SSTable inside a cache.  It exists (only) to allow tests to replace its
+// closer with a testable fake.
+type Item interface {
+	GetSSTable() *sstable.Reader
+	Close() error
+}
+
+type item struct {
+	r *sstable.Reader
+}
+
+func (i *item) GetSSTable() *sstable.Reader {
+	return i.r
+}
+
+func (i *item) Close() error {
+	return i.r.Close()
+}
+
+func NewCache(p lru.ParamsWithDisposal, fs pyramid.FS, readerOptions sstable.ReaderOptions) cache {
```

Comment:
Not sure why the `ParamsWithDisposal` are passed here. Is it a smart way of passing the underline cache configuration? 
Why do we need another dispose hook for the layer that calls the cache?

Reply:
Not sure what you mean, you're way ahead of me.  It's a *stupid* way of giving the configuration parameters.  Anything on ParamsWithDisposal is fair game, including 	Name, Logger, Size, Shards, which I think the caller will have to set (probably from configuration).